dashboard/app/models/user.rb

```diff
@@ -620,8 +620,16 @@ def user_level_locked?(script_level, level)
   def next_unpassed_progression_level(script)
     user_levels_by_level = user_levels_by_level(script)
 
-    # Find the user level that we've most recently had progress on
-    user_level = user_levels_by_level.values.flatten.sort_by!(&:updated_at).last
+    # some of our user_levels may be for levels within level_groups, or for levels
+    # that are no longer in this script. we want to ignore those, and only look
+    # user_levels that have matching script_levels
+    levels = script.script_levels.map(&:level_ids).flatten
```

Comment:
You may be right. I know little about (and often forget about) this feature.

I guess in the case that a level appears in multiple script_levels, and it's the one you have the most recent progress on, we'll always treat the first of the script_levels containing the level as your current stage - even if that level is not currently active for the first script_level.

we could instead make this something like:
`levels = script.script_levels.map{|script_level| script_level.oldest_active_level.id}`
which I think would be more resilient to this?